### ISSUE TITLE
Uncaught RangeError: Invalid array length

### ISSUE THREAD

**Issue by akaifi** [steelbrain/linter-ui-default#410]
[Enter steps to reproduce:]

Update atom from 1.10 to 1,19 and I got alot of errors this is one of them

**Atom**: 1.19.0 x64
**Electron**: 1.6.9
**OS**: "Fedora release
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.6.4

### Stack Trace

Uncaught RangeError: Invalid array length

```
At /usr/share/atom/resources/app/src/text-editor-component.js:2785

RangeError: Invalid array length
    at TextEditorComponent.populateVisibleTiles (/usr/share/atom/resources/app/src/text-editor-component.js:2785:45)
    at TextEditorComponent.updateSyncBeforeMeasuringContent (/usr/share/atom/resources/app/src/text-editor-component.js:363:16)
    at TextEditorComponent.screenPositionForPixelPosition (/usr/share/atom/resources/app/src/text-editor-component.js:2274:18)
    at getBufferPositionFromMouseEvent (/packages/linter-ui-default/lib/editor/helpers.js:9:50)
    at /packages/linter-ui-default/lib/editor/index.js:168:29)
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

### Commands

```
     -3:27.1.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always)
```

### Non-Core Packages

```
api-docs 0.0.6 
atom-autocomplete-php 0.25.6 
atom-beautify 0.30.4 
atom-runner 2.7.1 
atom-ternjs 0.18.3 
auto-detect-indentation 1.3.0 
autocomplete-modules 1.6.10 
autocomplete-php 0.3.7 
autocomplete-python 1.10.0 
busy-signal 1.4.3 
caniuse 0.9.2 
color-picker 2.2.5 
docblockr 0.11.0 
file-icons 2.1.10 
flex-tool-bar 0.12.0 
git-plus 7.9.3 
highlight-selected 0.13.1 
intentions 1.1.5 
linter 2.2.0 
linter-jsonlint 1.3.0 
linter-php 1.3.2 
linter-phpcs 1.6.7 
linter-phpmd 1.5.3 
linter-sass-lint 1.7.4 
linter-shellcheck 1.4.4 
linter-ui-default 1.6.4 
minimap 4.29.6 
minimap-highlight-selected 4.6.1 
minimap-pigments 0.2.2 
open-in-browsers 0.0.30 
open-recent 5.0.0 
open-terminal-here 2.3.1 
php-twig 4.0.0 
pigments 0.40.2 
project-manager 3.3.5 
react 0.16.2 
Remote-FTP 1.1.3 
seti-syntax 1.1.3 
seti-ui 1.9.0 
svg-preview 0.11.0 
sync-settings 0.8.2 
todo-show 1.11.0 
tool-bar 1.1.0 
w3c-validation 0.4.0 
webbox-color 0.5.9 
wordpress-api 1.2.2 
```

**Comment by dflock** [comment]
Seems to happen when using the `split-diff` package and scrolling past the end of the file.

**Atom**: 1.20.0-beta0 x64
**Electron**: 1.6.9
**OS**: Ubuntu 16.04.3
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.6.4


### Stack Trace

Uncaught RangeError: Invalid array length

```
At /usr/share/atom-beta/resources/app/src/text-editor-component.js:2785

RangeError: Invalid array length
    at TextEditorComponent.populateVisibleTiles (/usr/share/atom-beta/resources/app/src/text-editor-component.js:2785:45)
    at TextEditorComponent.updateSyncBeforeMeasuringContent (/usr/share/atom-beta/resources/app/src/text-editor-component.js:363:16)
    at TextEditorComponent.screenPositionForPixelPosition (/usr/share/atom-beta/resources/app/src/text-editor-component.js:2274:18)
    at getBufferPositionFromMouseEvent (/packages/linter-ui-default/lib/editor/helpers.js:9:50)
    at /packages/linter-ui-default/lib/editor/index.js:168:29)
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

### Commands

```
     -2:23.1.0 intentions:highlight (input.hidden-input)
     -2:22.8.0 command-palette:toggle (input.hidden-input)
     -2:19.7.0 core:confirm (input.hidden-input)
     -2:19.7.0 editor:toggle-soft-wrap (input.hidden-input)
     -2:00.4.0 intentions:highlight (input.hidden-input)
     -2:00.1.0 command-palette:toggle (input.hidden-input)
     -1:57.1.0 core:confirm (input.hidden-input)
     -1:57.1.0 editor:toggle-soft-wrap (input.hidden-input)
     -1:15.6.0 tree-view:remove (span.name.icon.default-icon)
     -0:48.8.0 intentions:highlight (input.hidden-input)
     -0:48.6.0 application:open-file (input.hidden-input)
     -0:31.6.0 scroll-sync:toggle (a.inline-block.text-highlight)
     -0:26.4.0 split-diff:toggle (a.inline-block.text-highlight)
     -0:10.8.0 editor:move-to-first-character-of-line (input.hidden-input)
     -0:10.2.0 intentions:highlight (input.hidden-input)
     -0:10.1.0 core:move-to-top (input.hidden-input)
```

### Non-Core Packages

```
advanced-open-file 0.16.6 
ansible-snippets 0.2.0 
ariake-dark-syntax 0.1.3 
asciidoc-assistant 0.2.2 
asciidoc-image-helper 1.0.0 
asciidoc-preview 2.10.2 
atom-alignment 0.13.0 
Atom-Bash-Snippets 0.0.3 
atom-beautify 0.30.4 
atom-dark-fusion-syntax 2.2.0 
atom-jade 0.3.0 
atom-jinja2 0.6.0 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 1.0.0 
atom-meld 0.3.5 
atom-mermaid 3.1.0 
atom-monokai 0.10.9 
atom-special-char-highlighter 0.1.0 
autocomplete-ansible 0.1.13 
autocomplete-asciidoc 0.1.2 
autocomplete-bash-builtins 0.3.5 
autocomplete-html-entities 0.1.0 
autocomplete-modules 1.6.10 
autocomplete-project-paths 2.2.0 
autocomplete-python 1.10.0 
bang 2.1.0 
busy-signal 1.4.3 
character-table 0.4.3 
chmod 0.1.1 
color-picker 2.2.5 
context-git 0.17.3 
copy-as-rtf 0.9.3 
docblockr 0.11.0 
duotone-dark-earth-syntax 2.1.0 
duotone-dark-forest-syntax 2.1.0 
duotone-dark-sea-syntax 2.1.0 
duotone-dark-space-syntax 2.1.0 
duotone-dark-syntax 2.1.0 
easysort-yaml 0.4.1 
editorconfig 2.2.2 
emmet-simplified 2.5.0 
file-icons 2.1.10 
file-templates 0.2.0 
file-types 0.5.5 
file-watcher 1.1.0 
flatwhite-syntax 0.8.1 
format-shell 2.0.0 
genesis-syntax 1.0.9 
git-blame 1.2.1 
git-hide 1.0.0 
git-time-machine 1.5.9 
hard-wrap 1.1.0 
indent-guide-improved 1.4.13 
intentions 1.1.5 
json-converter 0.2.4 
language-ansible 0.2.1 
language-asciidoc 1.10.0 
language-babel 2.69.0 
language-batch 0.4.0 
language-csv 1.1.2 
language-diff 0.7.0 
language-docker 1.1.8 
language-dotfiles 1.1.0 
language-dots 0.0.4 
language-generic-config 1.4.0 
language-gradle 0.0.3 
language-groovy 0.7.0 
language-hosts 0.1.2 
language-ini 1.19.0 
language-latex 1.1.0 
language-log 1.8.0 
language-mermaid 0.6.0 
language-mongodb 0.3.1 
language-nginx 0.8.0 
language-powershell 4.0.0 
language-pug 0.0.21 
language-restructuredtext 1.1.0 
language-robot-framework 2.2.0 
language-svg 0.9.2 
language-vue 0.23.1 
last-cursor-position 0.9.2 
less-than-slash 0.17.0 
linter 2.2.0 
linter-alex 4.0.0 
linter-ansible-linting 1.3.0 
linter-ansible-syntax 1.2.0 
linter-csslint 1.3.4 
linter-docker 0.2.0 
linter-eslint 8.2.1 
linter-js-standard 3.9.3 
linter-js-yaml 1.2.7 
linter-jscs 4.1.2 
linter-jshint 3.1.5 
linter-jsonlint 1.3.0 
linter-less 2.6.0 
linter-markdown 5.1.0 
linter-proselint 3.2.1 
linter-pug 1.3.1 
linter-pycodestyle 2.0.2 
linter-pydocstyle 0.4.7 
linter-ruby 1.2.5 
linter-sass-lint 1.7.4 
linter-shellcheck 1.4.4 
linter-swagger 0.4.1 
linter-ui-default 1.6.4 
linter-write-good 0.9.0 
linter-xmllint 1.4.3 
MagicPython 1.0.11 
markdown-themeable-pdf 1.2.0 
markdown-writer 2.6.5 
minimap 4.29.6 
minimap-bookmarks 0.4.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-linter 2.0.2 
minimap-selection 4.5.0 
minimap-split-diff 0.3.7 
monokai 0.24.0 
monokai-slate 0.6.0 
nebula-syntax 0.4.4 
nebula-ui 0.6.0 
nord-atom-syntax 0.9.1 
nord-atom-ui 0.11.0 
oceanic-next 1.0.0 
oceanic-reef-syntax 0.9.0 
open-terminal-here 2.3.1 
pigments 0.40.2 
prettier-atom 0.37.0 
pretty-json 1.6.4 
pubster-syntax 3.0.1 
regex-railroad-diagram 0.19.3 
robot-selenium-snippet 1.0.1 
rst-preview-pandoc 0.1.12 
scroll-sync 0.2.3 
set-syntax 0.3.2 
slime 2.1.0 
sort-lines 0.14.0 
split-diff 1.4.3 
ssh-config 0.10.1 
standardjs-snippets 2.6.0 
Sublime-Style-Column-Selection 1.7.4 
susave 0.4.0 
symbols-tree-view 0.14.0 
synced-sidebar 0.4.3 
tablr 1.8.3 
tabs-closer 0.16.0 
text-manipulation 0.6.0 
title-case 0.4.0 
toggle-quotes 1.0.1 
tornado-templates 0.2.0 
tree-view-git-status 1.4.0 
url-open 0.1.0 
Zen 0.18.0 
```

**Comment by Arcanemagus** [comment]
Duplicate of https://github.com/atom/atom/issues/15247

**Comment by Arcanemagus** [comment]
Looks like this was fixed in Atom v1.21.0.
